pkg/controller/core/resource_controller.go

```diff
@@ -47,17 +47,41 @@ const (
 	errorSettingResourceBindStatus   = "Failed to set resource binding status"
 	errorResettingResourceBindStatus = "Failed to reset resource binding status"
 	waitResourceIsNotAvailable       = "Waiting for resource to become available"
-
-	// RequeueOnWait - requeue after duration when waiting for resource state
-	RequeueOnWait = 30 * time.Second
-	// RequeueOnSuccess - requeue after duration when resources status is successful
-	RequeueOnSuccess = 2 * time.Minute
 )
 
-// Commonly used reconciliation results.
+// Suggested reconcile results. A reconciliation that either altered the
+// requested Kubernetes object or returned an error is considered dirty.
+// Reconciliations that do not alter the object and do not return an error are
+// considered clean.
 var (
-	Result        = reconcile.Result{}
-	ResultRequeue = reconcile.Result{Requeue: true}
+	// RequeueIfDirty does not queue a new reconciliation. A reconciliation will
+	// be queued implicitly if the current reconciliation was dirty. Prefer
+	// RequeueIfDirty if either the current reconciliation is known to be dirty,
+	// or further reconciliation is not necessary until the requested object is
+	// altered outside of this controller.
+	RequeueIfDirty = reconcile.Result{Requeue: false}
+
+	// RequeueNow queues a new reconciliation immediately. Beware that immediate
+	// requeues experience exponential backoff; the object under reconciliation
+	// will be reconciled with decreasing frequency until a reconciliation
+	// returns RequeueIfDirty or RequeueLater (and no error). Prefer RequeueNow
+	// if the reconciler encounters, but does not return, an error and would
+	// like to retry reconciliation as soon as possible.
+	RequeueNow = reconcile.Result{Requeue: true}
+
+	// RequeueSoon queues a new reconciliation to be processed after a short
+	// delay. Doing so will preempt any reconciliations queued further into the
+	// future, for example due to exponential backoff. Prefer RequeueSoon when
+	// actively waiting for an external system to sync with the specified state
+	// of the requested object.
+	RequeueSoon = reconcile.Result{RequeueAfter: 30 * time.Second}
+
+	// RequeueLater queues a new reconciliation to be processed after a moderate
+	// delay. Doing so will preempt any reconciliations queued further into the
+	// future, for example due to exponential backoff. Prefer RequeueLater when
+	// _not_ actively waiting for an external system to sync with the specified
+	// state of the requested object.
+	RequeueLater = reconcile.Result{RequeueAfter: 2 * time.Minute}
```

Comment:
As I have mentioned earlier, I don't think `30 sec` or `2  Min` will (or even should) work for all types. Thus, it seems a bit too prescriptive.

Reply:
In practice we already have 2 minute and 30 second constants that (I believe) you added to this file. Why were they okay in that case but not here?

Reply:
I understand they are not ideal in every case - they are suggestions. I also considered making this a function that took a duration.

Reply:
TBH, I am not sure that 30sec and 2 min are correct values, to begin with.